tools/autograd/templates/VariableType.cpp

```diff
@@ -345,7 +339,7 @@ static variable_list flatten(const Tensor& x, const TensorList& y) {
   return r;
 }
 
-static variable_list flatten(const Tensor& x, const TensorList& y, const Tensor& z) {
+static variable_list flatten(const Tensor& x, TensorList y, const Tensor& z) {
```

Comment:
Why can't it be a `const&` anymore?

Reply:
ArrayRefs (TensorList, IntList) should be passed by value. From LLVM's ArrayRef docs, which we copied into ArrayRef.h:

```
This is intended to be trivially copyable, so it should be passed by value
```